### PR TITLE
[tizen/nnfw] Add invoke for nnfw

### PR DESCRIPTION
**Pull request by kparichay**
Added invoke implementation for nnfw along with some bug fixes

Refer #1677

**Self evaluation:**
1. Build test: [ ]Passed [ ]Failed [*]Skipped
2. Run test: [ ]Passed [ ]Failed [* ]Skipped

Signed-off-by: Parichay Kapoor <pk.kapoor@samsung.com>

### CODE REVIEW

**Comment by jijoongmoon** [comment]
I think this should be working differently depending on the hw to use and we have to hand over this hw information to nnfw (I'm not sure they have..). If it is, then have to conform it is run on that hw (may be GPU). And in order to get the hw, you could use filter property.

**Comment by kparichay** [comment]
> I think this should be working differently depending on the hw to use and we have to hand over this hw information to nnfw (I'm not sure they have..). If it is, then have to conform it is run on that hw (may be GPU). And in order to get the hw, you could use filter property.

Yes, that is the part of configuring `backends`, which is a NYI item. It will be added as a subsequent PR, once the base version starts to run.
However, the interface for invoke is same for all the HW types, just setting backend is required using the interface provided. 
I will the confirm the behavior after the invoke unit-test is complete.

**Comment by myungjoo** [comment]
When you start working on configuring hw backends of nnfw, please look after the consistency with other filter subplugins; i.e., tf-lite nnapi configurations. To keep the consistency, you may modify tf-lite side for hw backend configurations as well.

File: ext/nnstreamer/tensor_filter/tensor_filter_nnfw.c [hunk 1]
```diff
@@ -161,6 +161,79 @@ nnfw_getOutputDim (const GstTensorFilterProperties * prop,
...
+  g_return_val_if_fail (prop != NULL, -EINVAL);
+  g_return_val_if_fail (mem != NULL, -EINVAL);
+  g_return_val_if_fail (pdata->session != NULL, -EPERM);
+
+  if (is_input) {
+    g_return_val_if_fail (prop->input_configured == TRUE, -EPERM);
+    num_tensors = prop->input_meta.num_tensors;
+  } else {
+    g_return_val_if_fail (prop->output_configured == TRUE, -EPERM);
+    num_tensors = prop->input_meta.num_tensors;
```
Comment (dongju-chae) [inline ext/nnstreamer/tensor_filter/tensor_filter_nnfw.c]: Oh, it should be `prop->output_meta`.

Comment (kparichay) [reply ext/nnstreamer/tensor_filter/tensor_filter_nnfw.c]: that's big mistake, updated.

**Pull request merged**